Error reporter

danielwelch opened the pull request:
implementing pluggable error reporters, as discussed in #210.

I've proposed some implementations compatible with this for [rollbar](https://github.com/danielwelch/django-q-rollbar) and [sentry](https://github.com/danielwelch/django-q-sentry) with some documentation available in each repository's README.md.

Includes documentation changes for configuring error reporters.

joeyespo commented:
This looks great! 👍

Eagllus commented on docs/errors.rst:
```diff
@@ -0,0 +1,9 @@
+
+Error reporting plugins register a class which implements a ``report`` method, which is invoked when a Django Q cluster encounters an error, pasing information to the particular service. Error reporters must be :ref:`configured<error_reporter>` via the ``Q_CLUSTER`` dictionary in your :file:`settings.py`. These settings are passed as kwargs upon initiation of the Error Reporter. Therefore, in ord ...
+
+For example implementations, see `django-q-rollbar <https://github.com/danielwelch/django-q-rollbar>` and `django-q-sentry <https://github.com/danielwelch/django-q-sentry>`
```
I like that you provide examples except that they are references to a different repository.
Shouldn't `django-q-rollbar` be included by default in this repo?

danielwelch replied:
@Eagllus 
> Any idea's on how you would like to prevent this change from breaking everybody's current setup?

Understandable. To me, the most obvious and probably easiest solution is just to leave the old rollbar support in place along side the error reporter implementation. Then no one currently using rollbar support needs to modify their `Q_CLUSTER` settings, while new projects can instead use the `error_reporter` config setting and include the `django-q-rollbar` extra. Seems to me this would involve simply "putting back" what was removed from `django_q/conf.py` and `django_q/cluster.py`. I could modify the documentation changes to re-include the rollbar section, maybe with a note about that config setting being there for compatibility purposes and a link to the error_reporter setting. Maybe a deprecation warning if `Q_CLUSTER['rollbar'] is detected, if you're inclined.

> I like that you provide examples except that they are references to a different repository.
Shouldn't django-q-rollbar be included by default in this repo?

In my opinion, different repositories is a strength (and kind of the point). I don't see any reason that `django-q-rollbar` should be included by default over, let's say, `django-q-sentry` aside from the backwards compatibility issue you brought up, which I think is valid. So the question seems to be, what is the best way to provide rollbar support: via `Q_CLUSTER['rollbar']`" vs. including `django-q-rollbar` in the django-q repository itself. To me, it seems cleaner to provide `django-q-rollbar` in the same way that `django-q-sentry` and potentially other plugins are provided: via a seperate repository. But I am but a hopeful contributor and those design decisions are probably not mine to make.

danielwelch commented:
@Eagllus I went ahead and implemented the suggestion I made above and modified documentation as well. This should prevent anything from breaking for those already using the `rollbar` setting.

Koed00 commented:
Looks great @danielwelch . I agree that rollbar does not need to be included by default.
I initially added it cause that's what I use myself and I hoped some day someone would take it upon themselves to add support for other reporters.  Happy someone took the initiative.

Koed00 commented:
I merged the code to master and fixed some of links in the sphinx documentation. Looks good.
Before I do a full release, I still want to merge some more stuff and run a some tests on my own servers.
